Utils/Dataflow/015_CDSDocuments/getCDSDocuments.py

```diff
@@ -34,355 +34,393 @@
+                                 " for unauthenticated user. Specify"
+                                 " login/password or use Kerberos"
+                                 " authentication.\n")
+            else:
+                sys.stderr.write("JSON decoding failed.\n")
+            results = None
+
+    else:
+        results = None
+
+    if type(results) == list:
+        if len(results) > 1:
+            sys.stderr.write("(WARN) Supporting document search returned more"
+                             " than one result (%s)\n"
+                             "(WARN) Will be taken the first of the list\n"
+                              % str(len(results)))
+        results = results[0]
+    if type(results) != dict:
+        results = None
+
+    return results
 
 def formOutputData(GLANCEdata, ppCDSdata, sdCDSdata):
-  """
-  Combine input (GLANCE) information with received from CDS;
-  Generate dkbID for documents
-  Parameters:
-    GLANCEdata -- main input JSON data
-    ppCDSdata  -- paper information from CDS
-    sdCDSdata  -- supporting documents information from CDS:
-                  { $sdGlanceID : [ ... ], ... }
-  """
-  if type(ppCDSdata) != type(GLANCEdata) != type(sdCDSdata) != dict:
-    sys.stderr.write("(ERROR) formOutputData() expected parameters of type %s (get %s, %s, %s)\n"
-                                                          % (dict, type(GLANCEdata), type(ppCDSdata), type(sdCDSdata)))
-  result = {}
-
-  ppGLANCEdata = GLANCEdata.copy()
-  if ppGLANCEdata.get("supporting_notes"):
-    sdGLANCEdata = ppGLANCEdata.pop("supporting_notes")
-  else:
-    sdGLANCEdata = []
-
-  result["GLANCE"] = ppGLANCEdata
-  result["CDS"] = ppCDSdata
-  result["dkbID"] = dkbID(result, dataType.DOCUMENT)
-
-  sd_results = []
-  sd_result = {}
-  result["supporting_notes"] = sd_result
-
-  if type(sdGLANCEdata) != list:
-    sys.stderr.write("(WARN) GLANCE info for supporting_notes supposed to be of type %s (get %s)\n"
-                                                         % (list, type(sdGLANCEdata)))
+    """ Combine input and found metadata; generate and add dkbID.
+
+    Parameters:
+      GLANCEdata -- main input JSON data
+      ppCDSdata  -- paper information from CDS
+      sdCDSdata  -- supporting documents information from CDS:
+                    { $sdGlanceID : [ ... ], ... }
+    """
+    if type(ppCDSdata) != type(GLANCEdata) != type(sdCDSdata) != dict:
+        sys.stderr.write("(ERROR) formOutputData() expected parameters of type"
+                         " %s (get %s, %s, %s)\n" % (dict, type(GLANCEdata),
+                         type(ppCDSdata), type(sdCDSdata)))
+    result = {}
+
+    ppGLANCEdata = GLANCEdata.copy()
+    if ppGLANCEdata.get("supporting_notes"):
+        sdGLANCEdata = ppGLANCEdata.pop("supporting_notes")
+    else:
+        sdGLANCEdata = []
+
+    result["GLANCE"] = ppGLANCEdata
+    result["CDS"] = ppCDSdata
+    result["dkbID"] = dkbID(result, dataType.DOCUMENT)
+
```

Comment:
We should unify the method of getting key value from dictionary. 
*.get(key) will be better?